testsuite/do_tests.sh

```diff
@@ -76,78 +88,87 @@ while test $# -gt 0 ; do
-# BSD: -E
-# other: -r
-#
-EXTENDED_REGEX_PARAM=r
-/bin/sh -c "echo 'hello' | sed -${EXTENDED_REGEX_PARAM} 's/[aeou]/_/g' "  >/dev/null 2>&1 || EXTENDED_REGEX_PARAM=E
-
-# source helpers to make their functions available
-. "$(dirname $0)/junit_xml.sh"
-. "$(dirname $0)/run_test.sh"
-
-TEST_BASEDIR="$(sli -c 'statusdict/prgdocdir :: =only')"
-TEST_OUTDIR="${TEST_OUTDIR:-$( pwd )/reports}"
-TEST_LOGFILE="${TEST_OUTDIR}/installcheck.log"
-TEST_OUTFILE="${TEST_OUTDIR}/output.log"
-TEST_RETFILE="${TEST_OUTDIR}/output.ret"
-TEST_RUNFILE="${TEST_OUTDIR}/runtest.sh"
-
-if test -d "${TEST_OUTDIR}" ; then
-    rm -rf "${TEST_OUTDIR}"
+if test ! "${PREFIX}"; then
+    usage 2 "--prefix";
 fi
 
-mkdir "${TEST_OUTDIR}"
-
-PYTHON="${PYTHON:-python3}"
-
-TMPDIR="${TMPDIR:-${TEST_OUTDIR}}"
-TEST_TMPDIR="$(mktemp -d "${TMPDIR:-/tmp}/nest.XXXXX")"
-NEST_DATA_PATH="${TEST_TMPDIR}"
-export NEST_DATA_PATH
+if test ! "${REPORTDIR}"; then
+    usage 2 "--report-dir";
+fi
 
+if test "${PYTHON}"; then
+    command -v nosetests >/dev/null 2>&1 || {
+        echo "Error: PyNEST testing requested, but command 'nosetests' cannot be executed."
+        exit 1
+    }
+    NOSE="$(command -v nosetests)"
+fi
 
-# Check for old version of the /mpirun command, which had the NEST executable hardcoded
-if test "x$(sli -c '/mpirun load cva_t Flatten length 3 eq =')" = xtrue ; then
-    echo "  Unable to run tests because you compiled with MPI and ~/.nestrc contains"
-    echo "  an old definition of /mpirun. If you were using the standard definition,"
-    echo "  please replace it by"
-    echo
-    echo "  /mpirun"
-    echo "  [/integertype /stringtype /stringtype]"
-    echo "  [/numproc     /executable /scriptfile]"
-    echo "  {"
-    echo "   () ["
-    echo "    (mpirun -np ) numproc cvs ( ) executable ( ) scriptfile"
-    echo "   ] {join} Fold"
-    echo "  } Function def"
-    echo
-    echo "  If you used a custom definition, please adapt it so that the signature"
-    echo "  of your version matches the one above (i.e. taking number of processes,"
-    echo "  executable and scriptfile as arguments; the old one just took number of"
-    echo "  processes and slifile, the executable \"nest\" was hard-coded)."
-    echo
-    echo
+python3 -c "import junitparser" >/dev/null 2>&1
+if test $? != 0; then
+    echo "Error: Required Python package 'junitparser' not found."
     exit 1
 fi
 
+# source helpers to set environment variables and make functions available
+. "${PREFIX}/bin/nest_vars.sh"
+. "$(dirname $0)/junit_xml.sh"
+. "$(dirname $0)/run_test.sh"
 
-# Remember: single line exports are unportable!
+if test -d "${REPORTDIR}"; then
+    rm -rf "${REPORTDIR}"
+fi
+mkdir "${REPORTDIR}"
 
-NEST_BINARY=nest_serial
+TEST_BASEDIR="${PREFIX}/share/doc/nest"
+TEST_LOGFILE="${REPORTDIR}/installcheck.log"
+TEST_OUTFILE="${REPORTDIR}/output.log"
+TEST_RETFILE="${REPORTDIR}/output.ret"
+TEST_RUNFILE="${REPORTDIR}/runtest.sh"
+TEST_TMPDIR=$(mktemp -d "${REPORTDIR}/nest.XXXXX")
```

Comment:
Since we always set up a clean `REPORTDIR`,  is there any need to have a randomly numbered `TMPDIR`?

Reply:
Good catch! Actually it seems like we don't even use `TEST_TMPDIR` anywhere. I've thus removed the line in question in 76cf66d4c.